.github/workflows/ci-fast.yml

```diff
@@ -82,29 +82,79 @@ jobs:
       python-version: '3.9'
       os: ubuntu-latest
     secrets: inherit
+  linting:
+    if: github.event.pull_request.draft == false
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.12']
+      fail-fast: false
+    uses: ./.github/workflows/linting.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      os: ${{ matrix.os }}
+    secrets: inherit
   custom-ubuntu-setup-and-unit-test:
+    needs: linting
     if: github.event.pull_request.draft == false
     strategy:
       matrix:
-        os: [arc-runner-set]
-        python-version: ['3.10']
+        os: [ubuntu-latest]
+        python-version: ['3.12']
```

Comment:
I'd better use `3.11` as default, since it has 4 times more usage according to the public data I was able to find. `3.10` is even higher, but also slower

Reply:
hmm i think we need not only public data but also zenml usage, I will check that but set 3.11 by default for now